src/App.tsx

```diff
@@ -8,53 +10,179 @@
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [showPostLoader, setShowPostLoader] = useState<boolean>(false);
+  const [showCommentLoader, setShowCommentLoader] = useState<boolean>(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isSidebarVisible, setIsSidebarVisible] = useState<number | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [hasPostError, setHasPostError] = useState<boolean>(false);
+  const [hasCommentError, setHasCommentError] = useState<boolean>(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(res => {
+        setUsers(res as User[]);
+      })
+      .catch(() => new Error('Failed to fetch users'));
+  }, []);
 
-              <Loader />
+  const handleUserPosts = (user: User) => {
+    setShowPostLoader(true);
+    getUserPosts(user.id)
+      .then(res => {
+        setUserPosts(res as Post[]);
+        setHasPostError(false);
+      })
+      .catch(() => {
+        setHasPostError(true);
+      })
+      .finally(() => {
+        setShowPostLoader(false);
+      });
+  };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const handleShowComment = (currentPost: Post) => {
+    setIsSidebarVisible(currentPost.id);
+    setShowCommentLoader(true);
+
+    setSelectedPost(currentPost);
+
+    getUserPostComments(currentPost.id)
+      .then(res => {
+        setPostComments(res as Comment[]);
+        setHasCommentError(false);
+      })
+      .catch(() => {
+        setHasCommentError(true);
+      })
+      .finally(() => {
+        setShowCommentLoader(false);
+      });
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    deleteComment(commentId)
+      .then(() => {
+        setPostComments(prevPostComments =>
+          prevPostComments.filter(
+            (comment: Comment) => comment.id !== commentId,
+          ),
+        );
+      })
+      .catch(() => new Error('Failed to delete comment'));
+  };
+
+  const addComment = (newComment: CommentData) => {
+    setIsAddingComment(true);
+    addNewComment(newComment)
+      .then(res => {
+        setPostComments([...postComments, res as Comment]);
+        setIsAddingComment(false);
+      })
+      .catch(() => new Error('Failed to add comment'));
+  };
```

Comment:
The method naming convention you're using here doesn't align with the best practices. The method name `addComment` sounds more like a method that would handle an event related to adding a comment, such as a click event on an add button. But in this case, it's adding a comment. A more appropriate name might be `createComment` or `submitComment`. This makes it clear what the method does just by looking at its name.